webapp/components/ErrorBoundary.jsx

```diff
@@ -0,0 +1,134 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+// This is being pulled form the CDN currently
+// import Raven from 'raven-js';
+
+import Logo from '../assets/Logo';
+import {Error404} from '../errors';
+
+class ErrorBox extends Component {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    subtext: PropTypes.string
+  };
+
+  render() {
+    return (
+      <div style={{margin: 'auto', maxWidth: 600}}>
+        <div
+          style={{
+            border: '10px solid #D6D4EA',
+            margin: 20,
+            padding: '20px 40px 20px'
+          }}>
+          {this.props.subtext
+            ? <h1 style={{opacity: 0.2, float: 'right'}}>
+                {this.props.subtext}
+              </h1>
+            : null}
+          <h1>
+            {this.props.title}
+          </h1>
+          {this.props.children}
+        </div>
+        <div style={{textAlign: 'center'}}>
+          <a href="/">
+            <Logo size="30" />
+          </a>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {error: null};
+  }
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({error});
+    if (window.Raven) {
+      window.Raven.captureException(error, {extra: errorInfo});
+      window.Raven.lastEventId() && window.Raven.showReportDialog();
```

Comment:
didn't actually confirm here, but im pretty sure raven is synchronous and this API is how we document it